### PR TITLE
docs: Docs for open_mapped_doc, get_mapped_doc and sync

### PR DESCRIPTION
**Pull request by hasnain2808**
docs for open_mapped_doc, get_mapped_doc and model.sync. These functions are used at multiple places in erpnext to sync data between documents

1. get_mapped_doc
Screenshot from 2020-09-30 21-03-40

2. open_mapped_doc
Screenshot from 2020-09-30 21-04-07
gif:
js-utils-open-mapped-doc

3. model.sync
Screenshot from 2020-09-30 21-04-02

### CODE REVIEW

File: frappe_docs/www/docs/user/en/api/js-utils.md [hunk 1]
```diff
@@ -81,3 +81,28 @@ frappe.require(['/assets/frappe/chat.js', '/assets/frappe/chat.css'], () => {
...
 ```
+
+## frappe.model.open\_mapped\_doc
+`frappe.model.open_mapped_doc({method,frm})`
+
+Load the mapped document returned by the method in a new document.
+
+```js
+// open a new dunning document using the sales invoice data
+create_dunning: function(frm) {
```
Comment (scmmishra) [inline frappe_docs/www/docs/user/en/api/js-utils.md:92]: Can you use a frappe specific example?

**Review by scmmishra — CHANGES_REQUESTED** [review]
Can you add a screenshot in the description?

File: frappe_docs/www/docs/user/en/api/js-utils.md (hunk 1 above)
Comment (hasnain2808) [reply frappe_docs/www/docs/user/en/api/js-utils.md]: Done

File: frappe_docs/www/docs/user/en/api/js-utils.md [hunk 2]
```diff
@@ -81,3 +81,49 @@ frappe.require(['/assets/frappe/chat.js', '/assets/frappe/chat.css'], () => {
...
 ```
+
+## frappe.model.open\_mapped\_doc
+`frappe.model.open_mapped_doc({method,frm})`
+
+Load the mapped document returned by the method in a new document.
+
+```js
+// open a new Movie Review document form using the Movie Review document
+create_movie_review: function(frm) {
```
Comment (scmmishra) [inline frappe_docs/www/docs/user/en/api/js-utils.md:92]: Continue on the library management application example

File: frappe_docs/www/docs/user/en/api/js-utils.md [hunk 3]
```diff
@@ -81,3 +81,49 @@ frappe.require(['/assets/frappe/chat.js', '/assets/frappe/chat.css'], () => {
 	// chat.js and chat.css are loaded
 })
 ```
+
+## frappe.model.open\_mapped\_doc
```
Comment (scmmishra) [inline frappe_docs/www/docs/user/en/api/js-utils.md:85]: Also explain what mapped_doc does

File: frappe_docs/www/docs/user/en/api/js-utils.md [hunk 4]
```diff
@@ -81,3 +81,49 @@ frappe.require(['/assets/frappe/chat.js', '/assets/frappe/chat.css'], () => {
...
+}
+```
+
+![frappe.model.open\_mapped\_doc](/docs/assets/img/api/js-utils-open-mapped-doc.gif)
+*frappe.model.open\_mapped\_doc*
+
+## frappe.model.sync
+`frappe.model.sync(r)`
+
+Extract docs, docinfo (attachments, comments, assignments) from incoming request and set in `locals` and `frappe.model.docinfo`
```
Comment (scmmishra) [inline frappe_docs/www/docs/user/en/api/js-utils.md:108]: Needs more explanantion

**Review by scmmishra — CHANGES_REQUESTED** [review]

**Pull request closed without merging**